app/main.py

```diff
@@ -1,3 +1,82 @@
+from typing import Union
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: int) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: Union["Distance", int, float]) -> "Distance":
+        if isinstance(other, Distance):
+            return Distance(self.km + other.km)
+        elif isinstance(other, (int, float)):
+            return Distance(self.km + other)
+        else:
+            raise TypeError("Unsupported type for addition")
+
+    def __iadd__(self, other: Union["Distance", int, float]) -> "Distance":
+        if isinstance(other, Distance):
+            self.km += other.km
+        elif isinstance(other, (int, float)):
+            self.km += other
+        else:
+            raise TypeError("Unsupported type for in-place addition")
+        return self
+
+    def __mul__(self, other: Union[int, float]) -> "Distance":
+        if isinstance(other, (int, float)):
+            return Distance(self.km * other)
+        else:
+            raise TypeError("Unsupported type for multiplication")
+
+    def __truediv__(self, other: Union[int, float]) -> "Distance":
+        if isinstance(other, (int, float)):
+            new_km = round(self.km / other, 2)
+            return Distance(new_km)
+        else:
+            raise TypeError("Unsupported type for division")
```

Comment:
In the `__truediv__` method, there should be a check to ensure that `other` is not zero before performing the division. Dividing by zero will raise a `ZeroDivisionError`, so it's important to handle this case explicitly.